process/interceptors/processor/peerAuthenticationInterceptorProcessor.go

```diff
@@ -52,11 +65,23 @@ func (paip *peerAuthenticationInterceptorProcessor) Validate(_ process.Intercept
 
 // Save will save the intercepted peer authentication inside the peer authentication cacher
 func (paip *peerAuthenticationInterceptorProcessor) Save(data process.InterceptedData, fromConnectedPeer core.PeerID, _ string) error {
-	interceptedPeerAuthenticationData, ok := data.(interceptedDataMessageHandler)
+	interceptedPeerAuthenticationData, ok := data.(interceptedPeerAuthenticationMessageHandler)
 	if !ok {
 		return process.ErrWrongTypeAssertion
 	}
 
+	payloadBuff := interceptedPeerAuthenticationData.Payload()
```

Comment:
by checking the hardfork payload here means that the hardfork public key should be a registered validator, which is not the case. We need an exception for the intercepted peer auth message coming the registered public key from the config.
That is why I have left the TODO on L211 in process/heartbeat/interceptedPeerAuthentication.go